### PR TITLE
Fix DNS Callback - DNS Networking UT for coverage

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Fix DNS Callback - DNS Networking UT for coverage

Description
-----------
Add UT for IPv6 branch for 100% coverage 

Test Steps
-----------
Normal Unit Tests compilation method

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c [hunk 1]
```diff
@@ -132,6 +132,40 @@ void test_CreateSocket_success( void )
...
+void test_BindSocket_fail( void )
+{
+    struct freertos_sockaddr xAddress;
+    Socket_t xSocket;
+    uint16_t usPort;
+    uint32_t ret;
+
+    FreeRTOS_bind_ExpectAnyArgsAndReturn( pdFALSE );
+
+    ret = DNS_BindSocket( xSocket, usPort );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c:147]: Here is the definition of FreeRTOS_bind return value:
```
 * @return The return value is 0 if the bind is successful.
 *         If some error occurred, then a negative value is returned.
```

So pdFALSE/pdTRUE is not the return value of FreeRTOS_bind. Neither do DNS_BindSocket.

File: test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c [hunk 2]
```diff
@@ -132,6 +132,40 @@ void test_CreateSocket_success( void )
...
+void test_BindSocket_success( void )
+{
+    struct freertos_sockaddr xAddress;
+    Socket_t xSocket;
+    uint16_t usPort;
+    uint32_t ret;
+
+    FreeRTOS_bind_ExpectAnyArgsAndReturn( pdTRUE );
+
+    ret = DNS_BindSocket( xSocket, usPort );
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c:164]: Same as previous comment.

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c (hunk 1 above)
Comment (kar-rahul-aws) [reply test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c:147]: Updated.

File: test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c (hunk 2 above)
Comment (kar-rahul-aws) [reply test/unit-test/FreeRTOS_DNS_Networking/FreeRTOS_DNS_Networking_utest.c:164]: Updated

**Review by ActoryOu — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Pull request merged**